### PR TITLE
Scatterplots work with temporal YYYY-MM-DD scales

### LINKED ISSUE #1835: Temporal scales don't work in scatterplots
### Current Behavior

Colourings with a temporal scale (except for `num_date`) do not work in scatterplots, with NaNs appearing in the SVG coordinates, e.g.:
```
Error: <path> attribute d: Expected number, "MNaN 472.09999999…".
```

They should [behave similarly to how `num_date` currently does](https://nextstrain.org/zika?c=num_date&l=scatter&m=div), however all nodes without metadata values (usually all internal nodes and perhaps some tips) should not appear. Scatterplots already have the ability to hide nodes which don't have appropriate metadata ([e.g.](https://nextstrain.org/zika?c=author&l=scatter&m=div&scatterX=region)).

This was first identified in #1832 but it predates that PR.

### How to reproduce

1. Create a dataset with a temporal scale (values can be numerical or, if you're using a version of Auspice more recent than c5857225f2911091fac20395e091031c79683b11 they can be YYYY-MM-DD)

2. View a scatterplot with the temporal scale as the x or y axis.

### PR DESCRIPTION
**Pull request by jameshadfield**
Closes #1835

This code would be greatly enhanced by using TypeScript

- [ ] Checks pass
- [ ] If making user-facing changes, add a message in [CHANGELOG.md](https://github.com/nextstrain/auspice/blob/HEAD/CHANGELOG.md) summarizing the changes in this PR
- [ ] (to be done by a Nextstrain team member) [Create preview PRs on downstream repositories][1].

### CODE REVIEW

**Comment by jameshadfield** [comment]
@huddlej you can see your (formerly broken) example at https://nextstrain-s-nextstrain-smwbzp.herokuapp.com/staging/rabies/YYYY-MM-DD-coloring?l=scatter&scatterX=date&scatterY=div

File: src/util/scatterplotHelpers.js [hunk 1]
```diff
@@ -97,10 +97,14 @@ export function getFirstMatchingScatterVariable(options, tryTheseFirst, notThisV
  */
 export function addScatterAxisInfo(scatterVariables, axis, controls, tree, metadata) {
   const axisVar = scatterVariables[axis];
+  console.log("addScatterAxisInfo::axisVar", axis, axisVar)
```
Comment (genehack) [inline src/util/scatterplotHelpers.js:100]: intentional? (asking because the linting is complaining.)

**Review by genehack — APPROVED** [review]

File: src/util/scatterplotHelpers.js (hunk 1 above)
Comment (jameshadfield) [reply src/util/scatterplotHelpers.js:100]: Ahh. Intentionally added, unintentionally neglected to remove.

Comment (jameshadfield) [reply src/util/scatterplotHelpers.js:100]: That's not why CI failed here (that's due to a 6-hour `npm ci` call timing out the action), but I agree with #1839 in that it should cause CI to fail.

Comment (jameshadfield) [reply src/util/scatterplotHelpers.js]: [dropped in force push](https://github.com/nextstrain/auspice/compare/681951aa9e7793db220cd3c9866a203a2ffcdac5..f5e388ef35cb700fafb5377be4f758d86a188f67)

**Comment by huddlej** [comment]
Nice! Thanks, @jameshadfield!

**Pull request merged**